src/actions/actions_process.ts

```diff
@@ -0,0 +1,23 @@
+import * as semver from "semver"
+import * as Hub from "../hub"
+import {ActionRequest} from "../hub"
+import "./index.ts"
+
+async function execute(req: any) {
+    const request = ActionRequest.fromJSON(req.body)
+    request.instanceId = req.instanceId
+    request.webhookId = req.webhookId
+    const userAgent = req.userAgent
+    if (userAgent) {
+        const version = userAgent.split("LookerOutgoingWebhook/")[1]
+        req.lookerVersion = semver.valid(version)
+    }
+    const action = await Hub.findAction(req.actionId, {lookerVersion: req.lookerVersion})
+    return await action.validateAndExecute(request)
+}
+
+process.on("message", (req) => {
+    const response = execute(req)
+    response.then((val) => {process.send!(val)})
```

Comment:
nit: could use async?

Reply:
I had to make execute async in order to use await in the message event. If there is a better way to wrap it I would love to hear some suggestions

Reply:
the `execute` in this file is already async?

you could also not await `action.validateAndExecute` and not make this `execute` async and since it returns a promise it should be awaitable